### PR TITLE
bitcoin: Release tracking PR: `v0.32.0-rc1` - oh yeah!

### PR DESCRIPTION
**Pull request by tcharding**
This PR is just the final patch, primarily the changelog. Pretty happy with my effort, check it out: https://github.com/tcharding/rust-bitcoin/blob/04-04-release-bitcoin-rc1/bitcoin/CHANGELOG.md

### CODE REVIEW

File: bitcoin/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,102 @@
...
+- In hardcoded BIP-32 derivation paths: Remove `m/` prefix requirement [#2451](https://github.com/rust-bitcoin/rust-bitcoin/pull/2451)
+
+## Crate smashing
+
+This release we continued our effort to split up the library into
+smaller crates AKA crate smashing.
+
+- Release new `bitcoin-units` crate [docs](https://docs.rs/bitcoin-units/0.1.1/bitcoin_units/)
+- Release new `base58ck` crate [docs](https://docs.rs/base58ck/0.1.0/base58ck/)
+- Release new `bitcoin-i` crate [docs](https://docs.rs/bitcoin-io/0.1.2/bitcoin_io/)
```
Comment (sanket1729) [inline bitcoin/CHANGELOG.md:22]: typo in `bitcoin-io`

File: bitcoin/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,102 @@
...
+
+- Require `io::BufRead` instead of `io::Read` in consensus decode trait [#2240](https://github.com/rust-bitcoin/rust-bitcoin/pull/2240)
+- Make `Payload` and `AddressEncoding` private [#1979](https://github.com/rust-bitcoin/rust-bitcoin/pull/1979)
+- Remove `bech32` from the public API [#2381](https://github.com/rust-bitcoin/rust-bitcoin/pull/2381)
+- Make from_hex inherent for byte-like types [#2491](https://github.com/rust-bitcoin/rust-bitcoin/pull/2491)
+- Remove the FromHexStr trait [#2492](https://github.com/rust-bitcoin/rust-bitcoin/pull/2492)
+
+## Error handling
+
+- Split `Prevouts` errors out into specific error types 2042
```
Comment (sanket1729) [inline bitcoin/CHANGELOG.md:92]: nit: Are you also going to link these? The standalone numbers look strange, we can also remove them completely.

**Review by sanket1729 — COMMENTED** [review]
wow, thanks for the effort in the detailled changelog

File: bitcoin/CHANGELOG.md (hunk 2 above)
Comment (tcharding) [reply bitcoin/CHANGELOG.md:92]: Drats, I missed these - this is what the looked like before I hit them with my emacs macro to fill in the links. This was an oversight, thanks for catching it.

File: bitcoin/CHANGELOG.md (hunk 1 above)
Comment (tcharding) [reply bitcoin/CHANGELOG.md:22]: Fixed, thanks man!

**Comment by tcharding** [comment]
If #2655 merges (and releases) I rekon we should add an update to the lockfiles (to pick up units `0.1.1`) either to this or as the first thing on the RC branch, might be a good test of the branching process we discussed @apoelstra, for everyone else, and to clarify I understand the process, we are going to branch off a `0.32.0` branch from the first release candidate tagged patch  then keep working on `master`, add changes for the `0.32.0` release to that branch and apply them to `master` as well.

**Comment by tcharding** [comment]
Flagging that `units v0.1.1` has not docs link on crates.io and no page at docrs - might be because the build infrastructure is still spinning since we only just published. Verify docs link here:
- https://crates.io/crates/bitcoin-units/0.1.1

And that this page exists:
- https://docs.rs/bitcoin-units/0.1.1/bitcoin_units/

**Review by sanket1729 — APPROVED** [review]
utACK 0073a17e2066b405aee59d0bb6012260b85c696e.

Let's go

**Comment by apoelstra** [comment]
There is trailing whitespace in CHANGELOG.md, but that's fine, I won't hold up the release for it.

File: bitcoin/CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,102 @@
...
+- Remove "no-std" feature [#2233](https://github.com/rust-bitcoin/rust-bitcoin/pull/2233)
+- Enforce displaying `Amount` with trailing zeros [#2604](https://github.com/rust-bitcoin/rust-bitcoin/pull/2604)
+
+## Things you will almost certainly hit when upgrading
+
+- Rename `Signature` field names [#2338](https://github.com/rust-bitcoin/rust-bitcoin/pull/2338)
+  - `sig` becomes `signature`
+  - `hash_ty` becomes `sighash_type`
+- Rename `txid` to `compute_txid` [#2366](https://github.com/rust-bitcoin/rust-bitcoin/pull/2366)
+- In hardcoded BIP-32 derivation paths: Remove `m/` prefix requirement [#2451](https://github.com/rust-bitcoin/rust-bitcoin/pull/2451)
```
Comment (apoelstra) [inline bitcoin/CHANGELOG.md:13]: In 0073a17e2066b405aee59d0bb6012260b85c696e:

I don't know that anybody will hit this. It's certainly not a "you will almost certainly hit" thing. Somebody would have filed an issue much sooner about the m-prefix if it was.

**Review by apoelstra — APPROVED** [review]
ACK 0073a17e2066b405aee59d0bb6012260b85c696e

**Comment by apoelstra** [comment]
Tagged and published.

File: bitcoin/CHANGELOG.md (hunk 3 above)
Comment (tcharding) [reply bitcoin/CHANGELOG.md:13]: Oh, I can move it. I hit it the miniscript upgrade and it was surprising.

Comment (tcharding) [reply bitcoin/CHANGELOG.md:13]: I'll use this as the first PR verifying I have the new branching strategy internalised correctly (I've explained it again below).

**Comment by tcharding** [comment]
@apoelstra  I created a branch `0.32.x` for fixes to the RCs as discussed. For others (and again making sure I have got it) all PRs that fix bugs in the RC releases will target the `0.32.x` branch and also be ported to `master`. Then we cut the `0.32.0` release from the `0.32.x` branch and keep the branch for later point releases (as is customary).

**Comment by apoelstra** [comment]
@tcharding yes -- except that we should merge things to master and then port them to `0.32.x` rather than the other way around. (Excepting things like "update version numbers in Cargo.toml" which would go to the `0.32.x` branch and never to master.)

**Comment by tcharding** [comment]
Added three new "port" lables.

**Pull request merged**